_posts/2024-03-20-infinispan-15-tracing.adoc

```diff
@@ -0,0 +1,135 @@
+
+* Allow to configure it both globally and on-cache basis using the standard Infinispan configuration.
+* Trace more event types: cluster, cross-site, cache-store and security.
+* Allow to select and configure, also at runtime, the span types that we want to produce and send to the collector.
+
+All the source code we're presenting on this article is taken from https://github.com/fax4ever/infinispan-demo/tree/main/infinispan-15-tracing[this demo project].
+
+== Configure tracing
+
+=== Global configuration
+
+To enable tracing globally you need as usual to define the `collector-endpoint`, but instead of using a system property
+it is possible to use the infinispan server configuration, in which we introduced a component to configure the tracing globally.
+
+[source,xml]
+----
+<infinispan xmlns:server="urn:infinispan:server:15.0">
+   <cache-container name="default">
+      <tracing collector-endpoint="http://jaeger-collector.infinispan-15-tracing.svc.cluster.local:4318" security="true" service-name="infinispan-service-tracing" />
+   </cache-container>
+</infinispan>
+----
+
+By default we support the tracing with the OpenTelemetry Protocol (OTLP),
+with the default exporter protocol `http/protobuf`, and not anymore `grpc`.
+For instance the default Jaeger port using this combination is the `4318`.
+
+We also activate the tracing of the security audit. This is a new tracing category we introduced with Infinispan 15 and it is defined at cache manager level (globally).
+In the demo this is the configuration passed to the Infinispan Helm Chart:
+
+[source,yaml]
+----
+infinispan:
+ cacheContainer:
+   tracing:
+     collector-endpoint: "http://jaeger-collector.infinispan-15-tracing.svc.cluster.local:4318"
+     service-name: infinispan-service-tracing
+     security: true
+----
+
+=== Cache configuration
+
+The other tracing categories are configurable at cache level, since they are cache scoped.
+For instance this is a possible cache configuration enabling the tracing on more categories:
+
+[source,yaml]
+----
+replicatedCache:
+  encoding:
+    key:
+      mediaType: "application/x-protostream"
+    value:
+      mediaType: "application/x-protostream"
+  indexing:
+    enabled: "true"
+    storage: "local-heap"
+    startupMode: "NONE"
+    indexedEntities:
+      - "fax.play.image"
+  tracing:
+    enabled: true
+    categories:
+      - "container"
+      - "persistence"
+      - "cluster"
+      - "x-site"
+----
+
+In this case the enabled tracing categories will be container, persistence, cluster and x-site.
+By default only the container category is enabled.
+
+==== Runtime enabling / disabling them
+
+Furthermore, it is possible to enable / disable them or even the entire tracing at runtime.
+
+For instance this is a piece of code using the HotRod client to enable at runtime the category `container` and `persistence`,
+disabling the others. A similar command can be invoked also with the REST interface or the command line interface.
+
```

Comment:
It is possible to update enabled tracing categories at runtime and programmatically using the Infinispan HotRod client or the REST API. 

The following example enables tracing for the `container `and `persistence` categories while disabling any other previously enabled categories: